Assets/Scripts/HUD/TimerCountdown.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.UI;
+
+public class TimerCountdown : MonoBehaviour
+{
+    public TextMeshProUGUI countdownTextDisplay;
+    public GameObject endMissionText;
+    public int secondsLeft = 5 * 60;
+    private bool takingAway = false;
+
+    void Start()
+    {
+        if (countdownTextDisplay != null) countdownTextDisplay.text = CountdownString();
+    }
+
+    void Update()
+    {
+        if (!takingAway && secondsLeft > 0)
+        {
+            StartCoroutine(TimerTake());
+        }
+
+        if (secondsLeft <= 0)
+        {
+            endMissionText.SetActive(true);
+            //TODO: wait few seconds
+            //TODO: switch to end of report scene
+        }
+    }
+
+    private IEnumerator TimerTake()
```

Comment:
For coroutines, that's kind of a weird implementation.. I would start it in the Start method (you can pause it at any point) and I would either put while (true) or while (secondsLeft > 0) as conditions.

In a perfect world a coroutine would look kinda like that (minus the .text change): 
![image](https://user-images.githubusercontent.com/54850409/158806674-8a7008e2-4001-4c3e-b6e2-f874794139c8.png)
 and now you could start it in the Start method
"takingAway" can be a misleading variable name, but you could probably totally get rid of it with just rearranging some things

I would recommend splitting up the timer and the UI in different classes (we already have a UIManager), but for this purpose it's small enough that it wouldn't matter too much.